### PR TITLE
cds: add cluster name to error message

### PR DESCRIPTION
**Pull request by ramaraochavali**
*Description*: Cluster errors does not include the cluster name. This PR fixes that.
*Risk Level*: Low
*Testing*: Low
*Docs Changes*: N/A
*Release Notes*: N/A
[Optional Fixes #Issue]
[Optional *Deprecated*:]

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@mattklein123  Thanks. Merged master.

File: source/common/upstream/cds_api_impl.cc [hunk 1]
```diff
@@ -88,7 +88,7 @@ void CdsApiImpl::onConfigUpdate(const ResourceVector& resources, const std::stri
         ENVOY_LOG(debug, "cds: add/update cluster '{}'", cluster_name);
       }
     } catch (const EnvoyException& e) {
-      exception_msgs.push_back(e.what());
+      exception_msgs.push_back(fmt::format("{}: {}", cluster_name, e.what()));
```
Comment (htuch) [inline source/common/upstream/cds_api_impl.cc]: TBH, I think this is all getting a little crazy. We're iterating over human readable strings, which might ultimately be parsed out by machines, see https://github.com/envoyproxy/envoy/issues/6026.

Ultimately what we want is https://github.com/envoyproxy/envoy/issues/5739. In the interim, I think we should have in both of these cases some proto structure resembling this, and we can spit out the JSON equivalent in the exception messages etc.

This probably needs a broader discussion, but we should be very careful about not forming strong dependencies on non-structured text error messages that are supposed to be indicating something that needs to be parsed again, this is an anti-pattern </rant>

CC @mattklein123 @fredlas @SecurityInsanity

**Comment by mattklein123** [comment]
> This probably needs a broader discussion, but we should be very careful about not forming strong dependencies on non-structured text error messages that are supposed to be indicating something that needs to be parsed again, this is an anti-pattern

I agree. IMO this is all temporary and we should explicitly make no guarantees whatsoever about any error messages that are not in a proto/structured format. We should probably update the breaking change guide to reflect this.

**Comment by SecurityInsanity** [comment]
FWIW as we've been building out a suitable first patch internally we've also been under the assumption that "error messages" currently are not under any structured format currently, and can change whenever. (and the solution for #6026 should be incremental, i.e. yes we'll start with a wrapper type, and then build into more structured errors over time).

**Pull request merged**